raiden/blockchain/events.py

```diff
@@ -145,49 +144,53 @@ def get_all_netting_channel_events(
     )
 
 
-def decode_event_to_internal(event):
-    """ Enforce the binary encoding of address for internal usage. """
-    data = event.event_data
+def decode_event_to_internal(abi, log_event):
+    """ Enforce the binary for internal usage. """
+    # Note: All addresses inside the event_data must be decoded.
 
-    if data['args'].get('channel_identifier'):
-        data['channel_identifier'] = data['args'].get('channel_identifier')
+    decoded_event = decode_event(abi, log_event)
 
-    # Note: All addresses inside the event_data must be decoded.
-    if data['event'] == EVENT_TOKEN_NETWORK_CREATED:
-        data['token_network_address'] = to_canonical_address(data['args']['token_network_address'])
-        data['token_address'] = to_canonical_address(data['args']['token_address'])
+    if not decoded_event:
+        raise UnknownEventType()
```

Comment:
If we can't decode the event the abi is wrong, either the abi and the smart contract are out-of-sync or we are interacting with the wrong contract, either way this is a serious bug and should crash the client, going forward is not safe in this case